### PR TITLE
Allow changing the save directory

### PR DESCRIPTION
**Pull request by Ghabry**
Add a new cli option --save-path, when set the save path is != project path.

Modified emscripten and Android to use this. Android should redirect read-only directories (Hello, Android >=4.4 !) now to a writable location (for save and config)

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -838,7 +832,7 @@ static BOOL CALLBACK MyMiniDumpCallback(PVOID,
 		case ThreadCallback:
 		case ThreadExCallback:
 			return true;
-		case MemoryCallback:
+		case MemoryCallback:-
```
Comment (carstene1ns) [inline src/player.cpp]: Stray colon?

Comment (Ghabry) [inline src/player.cpp]: hu, where is that coming from oO

**Comment by fdelapena** [comment]
output test failed:

    output: src/main_data.cpp:94: const string& Main_Data::GetProjectPath(): Assertion `project_path.empty()' failed.

**Comment by Ghabry** [comment]
And sorry @BlisterBoy for messing with the Android code, I only added the save handling part :D

File: builds/android/src/org/easyrpg/player/game_browser/GameBrowserHelper.java [hunk 2]
```diff
@@ -151,8 +152,25 @@ public static void launchGame(Context context, ProjectInformation project){
 		// Test again in case somebody messed with the file system
 		if (GameBrowserHelper.isRpg2kGame(new File(path))) {
 			Intent intent = new Intent(context, EasyRpgPlayerActivity.class);
+			ArrayList<String> args = new ArrayList<String>();
+			
 			// Path of game passed to PlayerActivity via intent "project_path"
+			// Command line passed via intent "command_line"
+			args.add("--project-path");
```
Comment (BlisterBoy) [inline builds/android/src/org/easyrpg/player/game_browser/GameBrowserHelper.java]: Ho @Ghabry , you made my day !!!

**Comment by BlisterBoy** [comment]
No problem, now I know how to pass argument to the Player (I want to continue my playthrough of a game but I didn't know how to use --test-play from the android port to avoid a bug :p).

The possibility to change save folder is an interesting features. It will be usefull with the standalone mode if we solve the problem of the ressources copy.

**Comment by Ghabry** [comment]
@BlisterBoy 
That command line passing is a new SDLActivity feature, got introduced to Player few days before christmas. So you couldn't know this one. But was really useful because I could remove some Android JNI code from native code.

**Comment by BlisterBoy** [comment]
That's really cool, having more control on the Player from the App unleashes some possibilities ^^.

**Comment by Ghabry** [comment]
I havn't programmed in java for years and the first mistake I made was comparing strings with == ~.~

Open for review...

**Comment by Ghabry** [comment]
I have only one device with extra sd card storage available for testing (my Samsung tablet) and the ExternalStorage.java code detects it :+1:

**Comment by Ghabry** [comment]
Okay that was my last one.

**Comment by carstene1ns** [comment]
> Will need some updates to transifex later.

The source file will be updated automatically by jenkins, then transifex should automatically strip unused strings from the translations, so the only work to be done is adding new strings to the translated files (actually transifex should add placeholders already^^).

**Comment by BlisterBoy** [comment]
The --encoding flag you used in GameBrowserHelper doesn't work for me.
At the start there is a red screen indicating "Error: Invalide encoding: auto"

**Comment by BlisterBoy** [comment]
If I removed this flag, everything works.

**Comment by Ghabry** [comment]
Docs updated and rebased :)

**Comment by fdelapena** [comment]
> The --encoding flag you used in GameBrowserHelper doesn't work for me.
At the start there is a red screen indicating "Error: Invalide encoding: auto"

@BlisterBoy is this still happening after the rebase?

**Comment by Ghabry** [comment]
BlisterBoy: You must ndk-build my branch of the Player. Native code got changed for handling "auto"

**Comment by BlisterBoy** [comment]
Yep it works :).

**Pull request merged**